test/components/EditableArrayCard.test.ts

```diff
@@ -37,29 +50,43 @@ describe('EditableArrayCard', () => {
     });
   });
 
-  [
-    { action: 'new', prop: onCreate },
-    { action: 'edit', prop: onSave },
-  ].forEach(({ action, prop }) => {
-    it(`Handles ${action}`, async () => {
-      const newValue = fakeTextShort(),
-        tester = await new Tester(EditableArrayCard, { props }).mount();
-
-      expect(tester.find('input#name').length).toBe(0);
-      await tester.click(`button.btn-${action}`);
-      await tester.refresh();
-      expect(tester.find('input#name').length).toBe(1);
-
-      expect(prop).not.toHaveBeenCalled();
-      await tester.changeInput('input#name', newValue);
-      await tester.submit();
-      expect(prop).toHaveBeenCalledWith({
-        ...(action === 'edit' ? model[0] : {}),
-        name: newValue,
-      });
-
-      await tester.refresh();
-      expect(tester.find('input#name').length).toBe(0);
+  it('Handles onCreate', async () => {
+    const newValue = fakeTextShort(),
+      tester = await new Tester(EditableArrayCard, { props }).mount();
+
+    await fillOutAndSubmit(tester, 'new', newValue);
+
+    expect(onCreate).toHaveBeenCalledWith({ name: newValue });
+
+    await tester.refresh();
+    expect(tester.find('input#name').length).toBe(0);
+  });
+
+  it('Handles onSave', async () => {
+    const newValue = fakeTextShort(),
+      tester = await new Tester(EditableArrayCard, { props }).mount();
+
+    await fillOutAndSubmit(tester, 'edit', newValue);
+
+    expect(onSave).toHaveBeenCalledWith({
+      ...model[0],
+      name: newValue,
     });
+
+    await tester.refresh();
+    expect(tester.find('input#name').length).toBe(0);
+  });
+
+  it('Handles onSuccess', async () => {
+    const newValue = fakeTextShort(),
+      tester = await new Tester(EditableArrayCard, { props: { ...props, onSuccess } }).mount();
+
+    await fillOutAndSubmit(tester, 'new', newValue);
+
+    expect(onCreate).toHaveBeenCalled();
+    expect(onSuccess).toHaveBeenCalled();
+
+    await tester.refresh();
+    expect(tester.find('input#name').length).toBe(0);
```

Comment:
This is essentially testing that the card is no longer in form mode. There are many reasons it might not be 0, like if there was a validation error, or an error in the submit code, or an error in the switch-back-to-card code.